R/sensitivity-tornado-plot.R

```diff
@@ -0,0 +1,304 @@
+  )
+
+  # list of plots ----------------------------
+
+  # create plot for each output path
+  lsPlots <- purrr::map(
+    .x = data %>% split(.$OutputPath),
+    .f = ~ .createTornadoPlot(
+      .x,
+      defaultPlotConfiguration = customPlotConfiguration
+    )
+  )
+
+  # print plots without producing warnings
+  suppressWarnings(purrr::walk(lsPlots, ~ print(.x)))
+}
+
+#' @keywords internal
+#' @noRd
+.createTornadoPlot <- function(data,
+                               defaultPlotConfiguration) {
+  # update data dependent plot configuration
+  plotConfiguration <- defaultPlotConfiguration$clone()
+  plotConfiguration <- .updatePlotConfiguration(
+    plotConfiguration,
+    list(title = unique(data$OutputPath))
+  )
+
+  # calculate x-axis breaks and limits -------
+  pLimits <- .calculateLimits(data$PercentChangePK)
+  if (isTRUE(pLimits[1] > 0.5)) pLimits[1] <- 0.5
+  if (isTRUE(pLimits[2] < 1.5)) pLimits[2] <- 1.5
+
+  pBreaks <- .calculateBreaks(data$PercentChangePK, m = 5)
+
+  # map each PK parameter to its own plot ----
+
+  plotList <- purrr::map(
+    unique(data$PKParameter),
+    ~ {
+      dataSubset <- dplyr::filter(data, PKParameter == .x)
+
+      # basic plot setup -------------------------
+
+      plot <- ggplot(
+        dataSubset,
+        aes(
+          x = ParameterPath,
+          y = PercentChangePK,
+          fill = as.factor(ParameterFactor)
+        )
+      ) +
+        geom_col(
+          color = "grey",
+          na.rm = TRUE
+        ) +
+        coord_flip()
+
+      plot <- plot +
+        scale_y_continuous(
+          limits = pLimits,
+          breaks = pBreaks
+        )
+
+      # finalize plot ----------------------------
+
+      # note: facet wrap on unique PK parameter to obtain facet titles
+      plot <- plot +
+        facet_wrap(~PKParameter, scales = "fixed") +
+        labs(
+          x = plotConfiguration$xLabel,
+          y = plotConfiguration$yLabel,
+          title = NULL,
+          fill = plotConfiguration$legendTitle
+        ) +
+        theme_light(
+          base_size = 11
+        ) +
+        theme(
+          legend.position = plotConfiguration$legendPosition,
+          panel.grid.minor = element_blank(),
+          text = element_text(size = 11),
+          axis.text.y = element_text(margin = margin(l = 20, unit = "pt"))
+        )
+
+      # apply color scales
+      if (is.null(plotConfiguration$linesColor)) {
+        plot <- plot + scale_fill_brewer(palette = "Set2")
+      } else {
+        pLevels <- levels(as.factor(data$ParameterFactor))
+        pColor <- plotConfiguration$linesColor[1:length(pLevels)]
+        names(pColor) <- pLevels
+        plot <- plot + scale_fill_manual(
+          values = colorspace::lighten(pColor, amount = 0.2)
+        )
+      }
+
+      return(plot)
+    }
+  )
+
+  # compile individual plots -----------------
+
+  plotPatchwork <- patchwork::wrap_plots(plotList, ncol = 1) +
```

Comment:
Why use patchwork if you already use facet_wrap() ?

Reply:
I switched to using patchwork in `sensitivitySpiderPlot()` as required parameters couldn't be implemented with previous code using `facet_wrap()`. Now I apply facets to a single factor to keep a consistent look and preserve facet headers. This approach is now uniform across all plotting functions.